tests/playwright/src/specs/kind.spec.ts

```diff
@@ -54,34 +68,107 @@ beforeEach<RunnerTestContext>(async ctx => {
+      await playExpect(resourcesPage.kindResources).not.toBeVisible();
+      await statusBar.installKindCLI();
+      await playExpect(statusBar.kindInstallationButton).not.toBeVisible();
+    });
+    test('Verify that Kind CLI is installed', async () => {
+      await navigationBar.openSettings();
+      await playExpect(resourcesPage.kindResources).toBeVisible();
+    });
+    test('Kind extension lifecycle', async () => {
+      const extensionsPage = new ExtensionsPage(page);
+      await navigationBar.openExtensions();
+      const kindExtension = await extensionsPage.getInstalledExtension('Kind extension', extensionLabel);
+      await playExpect
+        .poll(async () => await extensionsPage.extensionIsInstalled(extensionLabel), { timeout: 10000 })
+        .toBeTruthy();
+      await playExpect(kindExtension.status).toHaveText('ACTIVE');
+      await kindExtension.disableExtension();
+      await navigationBar.openSettings();
+      await playExpect(resourcesPage.kindResources).not.toBeVisible();
+      await navigationBar.openExtensions();
+      await kindExtension.enableExtension();
+      await navigationBar.openSettings();
+      await playExpect(resourcesPage.kindResources).toBeVisible();
+    });
   });
+  describe.skipIf(process.env.GITHUB_ACTIONS && process.env.RUNNER_OS === 'Linux')(
+    'Deploying a pod to a Kind cluster',
+    async () => {
+      test(
+        'Create a Kind cluster',
+        async () => {
+          await navigationBar.openSettings();
+          await playExpect.poll(async () => resourcesPage.kindClusterExists()).toBeFalsy();
+          const createKindClusterPage = await resourcesPage.openClusterCreationPage();
+          await createKindClusterPage.checkClusterProperties();
+          await createKindClusterPage.createKindCluster();
+          await playExpect.poll(async () => resourcesPage.kindClusterExists()).toBeTruthy();
+          await playExpect(resourcesPage.kindClusterStatus).toHaveText(KindCLusterState.Running);
+        },
+        { timeout: 120000 },
+      );
+      test('Check for resources added with the Kind cluster', async () => {
+        const containersPage = await navigationBar.openContainers();
+        await playExpect.poll(async () => containersPage.containerExists(kindContainerName)).toBeTruthy();
+        const containerDetailsPage = await containersPage.openContainersDetails(kindContainerName);
+        playExpect(await containerDetailsPage.getState()).toEqual(ContainerState.Running.toLowerCase());
+
+        const volumesPage = await navigationBar.openVolumes();
+        await playExpect
+          .poll(async () => await volumesPage.kindClusterVolumeExists(kindContainerName), { timeout: 10000 })
+          .toBeTruthy();
+      });
+      test('Validate the right Kubernetes context is chosen', async () => {
+        await statusBar.validateKubernetesContext(kubernetesKontex);
+      });
+      test('Create a Pod to be deployed to the Kind cluster', async () => {
```

Comment:
I would stop here, this could be added in another PR, otherwise this PR will grow too much.